Interaction between serde and strum that results in a compiler error

Hi!

I've noticed that using strum's derive `EnumDiscriminants` with some serde variant attributes, results in a compiler error.

I'm not sure of this is an issue in strum, in serde, or both.

Here's a program to reproduce the issue:

### Cargo.toml
```
[dependencies]
serde = { version = "1.0", features = ["derive"] }
strum = "0.16.0"
strum_macros = "0.16.0"
```

### src/main.rs
```
extern crate strum;
#[macro_use] extern crate strum_macros;
extern crate serde;

use std::str::FromStr;

use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize, Debug, EnumDiscriminants)]
#[serde(tag = "type")]
enum Message {
    #[serde(rename = "request" )]
    Request { id: String, method: String, params: u8 },
    #[serde(rename = "response" )]
    Response { id: String, result: u8 },
}

fn main() {
    println!("Hello, world!");
}
```

### Compiler error
```
error[E0658]: the attribute `serde` is currently unknown to the compiler and may have meaning added to it in the future
  --> src/main.rs:12:5
   |
12 |     #[serde(rename = "request" )]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/29642

error[E0658]: the attribute `serde` is currently unknown to the compiler and may have meaning added to it in the future
  --> src/main.rs:14:5
   |
14 |     #[serde(rename = "response" )]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/29642

warning: unused import: `std::str::FromStr`
 --> src/main.rs:5:5
  |
5 | use std::str::FromStr;
  |     ^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
```

### Additional info
The program compiles successfully if the lines ` #[serde(rename = "response" )]`  are removed.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
+* Only copy across `"doc", "cfg", "allow", "deny"` attributes from main enum variants to discriminant variants. [#73](https://github.com/Peternator7/strum/issues/73)
+
 ## 0.17.1
 
 * Fixed an issue caused by combining [#60](https://github.com/Peternator7/strum/pull/60) and [#76](https://github.com/Peternator7/strum/pull/76)

--- a/strum_macros/src/macros/enum_discriminants.rs
+++ b/strum_macros/src/macros/enum_discriminants.rs
@@ -4,6 +4,12 @@ use syn;
 
 use helpers::extract_meta;
 
+/// Attributes to copy from the main enum's variants to the discriminant enum's variants.
+///
+/// Attributes not in this list may be for other `proc_macro`s on the main enum, and may cause
+/// compilation problems when copied across.
+const ATTRIBUTES_TO_COPY: &[&str] = &["doc", "cfg", "allow", "deny"];
+
 pub fn enum_discriminants_inner(ast: &syn::DeriveInput) -> TokenStream {
     let name = &ast.ident;
     let vis = &ast.vis;
@@ -66,9 +72,9 @@ pub fn enum_discriminants_inner(ast: &syn::DeriveInput) -> TokenStream {
 
         // Don't copy across the "strum" meta attribute.
         let attrs = variant.attrs.iter().filter(|attr| {
-            attr.parse_meta()
-                .map(|meta| !meta.path().is_ident("strum"))
-                .unwrap_or(true)
+            ATTRIBUTES_TO_COPY
+                .iter()
+                .any(|attr_whitelisted| attr.path.is_ident(attr_whitelisted))
         });
 
         discriminants.push(quote! { #(#attrs)* #ident });

--- a/strum_tests/Cargo.toml
+++ b/strum_tests/Cargo.toml
@@ -7,4 +7,5 @@ authors = ["Peter Glotfelty <peglotfe@microsoft.com>"]
 strum = { path = "../strum", features = ["derive"] }
 strum_macros = { path = "../strum_macros", features = [] }
 clap = "2.33.0"
+enum_variant_type = "0.2.0"
 structopt = "0.2.18"

--- a/strum_tests/tests/enum_discriminants.rs
+++ b/strum_tests/tests/enum_discriminants.rs
@@ -1,6 +1,8 @@
 extern crate strum;
 #[macro_use]
 extern crate strum_macros;
+#[macro_use]
+extern crate enum_variant_type;
 
 use strum::IntoEnumIterator;
 
@@ -183,4 +185,36 @@ fn from_test_complex() {
 fn from_ref_test_complex() {
     let rara = Rara;
     assert_eq!(EnumIntoComplexVars::A, (&EnumIntoComplex::A(&rara)).into());
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Eq, PartialEq, EnumDiscriminants, EnumVariantType)]
+#[strum_discriminants(
+    name(VariantFilterAttrDiscs),
+    derive(Display, EnumIter, EnumString),
+    strum(serialize_all = "snake_case")
+)]
+enum VariantFilterAttr {
+    #[evt(derive(Clone, Copy, Debug))]
+    DarkBlack(bool),
+    #[evt(skip)]
+    BrightWhite(i32),
+}
+
+#[test]
+fn filter_variant_attributes_pass_through() {
+    use std::str::FromStr;
+
+    let discriminants = VariantFilterAttrDiscs::iter().collect::<Vec<_>>();
+    let expected = vec![
+        VariantFilterAttrDiscs::DarkBlack,
+        VariantFilterAttrDiscs::BrightWhite,
+    ];
+
+    assert_eq!(expected, discriminants);
+    assert_eq!("dark_black", VariantFilterAttrDiscs::DarkBlack.to_string());
+    assert_eq!(
+        VariantFilterAttrDiscs::DarkBlack,
+        VariantFilterAttrDiscs::from_str("dark_black").unwrap()
+    );
 }

